File: core/src/main/java/com/andiag/core/BaseApplication.java
package com.andiag.core;

import android.app.Application;
import android.support.annotation.Nullable;

import uk.co.chrisjenx.calligraphy.CalligraphyConfig;

public abstract class BaseApplication extends Application {

    @Nullable
    protected abstract String getApplicationFont();

    @Override
    public void onCreate() {
        super.onCreate();
        if (getApplicationFont() != null) {
            CalligraphyConfig.initDefault(new CalligraphyConfig.Builder()
                    .setDefaultFontPath(getApplicationFont())
                    .setFontAttrId(R.attr.fontPath)
                    .build());
        }
    }

}

File: core/src/main/java/com/andiag/core/BaseApplication.kt
package com.andiag.core

import android.app.Application

import uk.co.chrisjenx.calligraphy.CalligraphyConfig

abstract class BaseApplication : Application() {

    protected abstract val applicationFont: String?

    override fun onCreate() {
        super.onCreate()
        if (applicationFont != null) {
            CalligraphyConfig.initDefault(CalligraphyConfig.Builder()
                    .setDefaultFontPath(applicationFont)
                    .setFontAttrId(R.attr.fontPath)
                    .build())
        }
    }

}
